controllers/metallb_controller.go

```diff
@@ -160,91 +161,62 @@ func (r *MetalLBReconciler) SetupWithManager(mgr ctrl.Manager, bgpType string) e
 func (r *MetalLBReconciler) syncMetalLBResources(config *metallbv1beta1.MetalLB) error {
 	logger := r.Log.WithName("syncMetalLBResources")
 	logger.Info("Start")
-	data := render.MakeRenderData()
-
-	data.Data["SpeakerImage"] = os.Getenv("SPEAKER_IMAGE")
-	data.Data["ControllerImage"] = os.Getenv("CONTROLLER_IMAGE")
-	data.Data["FRRImage"] = os.Getenv("FRR_IMAGE")
-	data.Data["IsOpenShift"] = r.PlatformInfo.IsOpenShift()
-	data.Data["NameSpace"] = r.Namespace
-	data.Data["KubeRbacProxy"] = os.Getenv("KUBE_RBAC_PROXY_IMAGE")
-	data.Data["DeployKubeRbacProxies"] = os.Getenv("DEPLOY_KUBE_RBAC_PROXIES") == "true"
 
-	data.Data["LogLevel"] = metallbv1beta1.LogLevelInfo
-	if config.Spec.LogLevel != "" {
-		data.Data["LogLevel"] = config.Spec.LogLevel
-	}
-
-	var err error
-	data.Data["MetricsPort"], err = valueWithDefault("METRICS_PORT", 7472)
+	actionConfig := new(action.Configuration)
+	chartValueOpts := &values.Options{}
+	chartValues, err := chartValueOpts.MergeValues(getter.All(r.HelmEnvSettings))
 	if err != nil {
 		return err
 	}
-	data.Data["MetricsPortHttps"], err = valueWithDefault("HTTPS_METRICS_PORT", 27472)
+	err = patchPrometheusValues(chartValues, r.Client)
 	if err != nil {
 		return err
 	}
-	data.Data["FRRMetricsPort"], err = valueWithDefault("FRR_METRICS_PORT", 7473)
+	patchControllerValues(chartValues, config, r.PlatformInfo.IsOpenShift())
+	err = patchSpeakerValues(chartValues, config)
 	if err != nil {
 		return err
 	}
-	data.Data["FRRMetricsPortHttps"], err = valueWithDefault("FRR_HTTPS_METRICS_PORT", 27473)
+	// TODO: patch rbac proxy settings if needed.
+	client := action.NewInstall(actionConfig)
```

Comment:
If the chart won't change (and I think it won't), we can create the client only once (also, wrapping it in a function)